scipy/spatial/transform/rotation.py

```diff
@@ -129,3 +129,35 @@ def as_dcm(self):
             return dcm[0]
         else:
             return dcm
+
+    @classmethod
+    def from_dcm(cls, mat, scalar=True):
+        # For now, mat must be a 3D matrix
+        num_rotations = mat.shape[0]
+        quat = np.empty((num_rotations, 4))
+        if scalar:
+            for rot_num in range(num_rotations):
+                A = mat[rot_num]
+                trA = A.trace()
+                choice = np.argmax(np.array([A[0, 0], A[1, 1], A[2, 2], trA]))
+
+                if choice in [0, 1, 2]:
+                    quat[rot_num, choice] = 1 - trA + 2 * A[choice, choice]
```

Comment:
I'm not 100% sure how to write this code in the "best way", but I probably would introduce 3 indices `i, j, k`, like:
```
i = choise
j = (choice + 1) % 3
k = (choice + 2) % 3
```
And then use this indices to write the formulas. The "idea" here is to avoid somewhat confusing names like `next_ind` and `next_next` and to use neutral mathematical indices. I think it will read slightly nicer. You might have another better idea.

By the way, this is a great generalization of 3 out of 4 cases.